### PR TITLE
Mepts c2

### PR DESCRIPTION
**Pull request by ningosi**
C2: Number of patients who initiated Pre-TARV during the current month and started TPI

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 1]
```diff
@@ -598,6 +638,50 @@ private CohortDefinition getPatientsWithTransferFromOtherHF() {
...
+    List<Concept> concepts = new ArrayList<>();
+    cd.addSearch("transferBasedOnObsDate", map(getPatientsTransferBasedOnObsDate(), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},locationList=${locationList}"));
+    cd.addSearch("getTypeOfPatientTransferredFrom", map(getTypeOfPatientTransferredFrom(), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},locationList=${locationList}"));
+    cd.addSearch("inProgramState", map(genericCohortQueries.getPatientsBasedOnPatientStates(hivMetadata.getHIVCareProgram().getProgramId(),
+            hivMetadata.getPateintTransferedFromOtherFacilityWorkflowState().getProgramWorkflowStateId()), "startDate=${onOrAfter},endDate=${onOrBefore},location=${locationList}"));
+    cd.setCompositionString("transferBasedOnObsDate OR getTypeOfPatientTransferredFrom OR inProgramState");
+    return cd;
+  }
+
+  public CohortDefinition getAdditionalExclusionCriteriaForC2() {
```
Comment (Edgencio) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: Looks Like the **getAdditionalExclusionCriteriaForC2()** is similar to **getAdditionalExclusionCriteriaForC1()**, maybe you could create one function with a generic name and reuse it#

**Review by Edgencio — CHANGES_REQUESTED** [review]

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -1850,6 +1850,15 @@
 		</description>
 	</globalProperty>
 
+	<!--6275-->
+	<globalProperty>
+		<property>eptsreports.hivCareConceptUuid</property>
```
Comment (Edgencio) [inline omod/src/main/resources/config.xml]: There is a duoplication of this concept... More info about this, on the change requests for the **MEPTS_C1** Pull Request.

**Review by Edgencio — CHANGES_REQUESTED** [review]

File: omod/src/main/resources/config.xml (hunk 2 above)
Comment (ningosi) [reply omod/src/main/resources/config.xml]: If C1 is merged, all these will be sorted out

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java (hunk 1 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: The difference comes in the parameter manipulation, one needs one month be sutracted from startDate for both 2 searches and one just maintain. I would suggest those to stay that way for now

**Comment by ningosi** [comment]
@cfaife this ticket relates to the one you just reviewed and merged. I will be fixing the conflicts and code refinement, then I have it be checked. The difference is so little compared to C1, then later we will finish with C3 before the entire ticket.

**Comment by ningosi** [comment]
@cfaife when you get time check on this one. I need to finish all the C sections

**Pull request merged**